### PR TITLE
Add Git Versioning Docs

### PR DESCRIPTION
**Pull request by brynnz22**
Issue #37 

This PR includes the documentation for releases. I am guessing this will be a first go at it and we will do releases, see how we like it, adjust and iterate. I set the release cycle as every first business day of the month if there are changes to `main`. We can change this how we like. For now, the `CHANGELOG.md` is manual. There are potential tools we can use to automate this, but I'm not sure that level of automation is necessary given the nature of this repo. The release part of Github auto-generates a description of the release anyway based on the merged requests. Right now I just include an outline in the `CHANGELOG`. We will have to update/erase some of it with our first release.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Does your PR link to an issue?
* [x] Have you described the changes this PR will make?

### New Notebook Submissions:

* [ ] Have you included a summary of the notebook in the README.md included updated links to the notebook?
* [ ] Does your PR include links to the new notebook **(in the branch)** for review using [nbviewer](https://nbviewer.jupyter.org/), [Colab](https://colab.research.google.com/), and [reviewnb](https://www.reviewnb.com/)? These three are the preferred ways to review changes and additions to notebooks during review.

### Notebook Fix Submissions:

* [ ] Does your PR include links to the updated notebook **(in the branch)** for review using [nbviewer](https://nbviewer.jupyter.org/), [Colab](https://colab.research.google.com/), and [reviewnb](https://www.reviewnb.com/)? These three are the preferred ways to review changes and additions to notebooks during review.

### CODE REVIEW

**Review by samobermiller — APPROVED** [review]

**Comment by brynnz22** [comment]
From Bea: Maybe make release date two weeks after nmdc-schema release to apply any new schema changes and keep in sync with new schema changes.

File: .github/RELEASE_GUIDE.md
Comment (kheal) [inline .github/RELEASE_GUIDE.md:8]: This is great.  One minor improvement would be to make sure each bullet is consistent stylistically.  Some start with capitalization, others do not.  Some end it periods, others do not.

File: CHANGELOG.md
Comment (kheal) [inline CHANGELOG.md:1]: I'm a little confused by this.  I think this document should be the actual change log, not just guidelines for how to write a change log?    So for this PR, I'd say we should have a changelog that corresponds to v1.0.0 .  Also I think we should explore using auto-generators for changelogs.  I never have, but let's try it out!  There are some good resources here: https://www.freecodecamp.org/news/a-beginners-guide-to-git-what-is-a-changelog-and-how-to-generate-it/

Comment (kheal) [reply CHANGELOG.md:1]: If we want to add guidelines for how to write/generate the change log, let's add them to the release guide - you've already got most of it.

Comment (brynnz22) [reply CHANGELOG.md:1]: @kheal there this outlines how the change log is managed and how links to how it is stylized based on standards that already exist. I advocate for leaving this here since it is very much part of the release process. I can change the heading to say `Change log guide` or something if that makes it clearer.

I did not see the point in auto-generating a changelog since the release will autogenerate the list of PRs that were merged. Many of those auto-generating changelogs will just create a list of all the commits which is uninformative to most readers, messy (especially if you are like me and commit every change you make). I was thinking a nice clean hand-curated change log is the way to go. Maybe we can discuss at our next meeting.

Comment (brynnz22) [reply CHANGELOG.md:1]: Sorry, that highlight portion is the actual change log. We don't have any changes yet since we don't have a release. I added it so we could all see how it is structured.

Comment (brynnz22) [reply CHANGELOG.md:1]: I can create a release as part of this PR so it makes sense.

Comment (kheal) [reply CHANGELOG.md:1]: This all makes sense.  I think if we change the name or location of the guidelines (I agree they are worthwhile to keep in the repo) and add the first entry for the CHANGELOG that'll be great.

**Comment by brynnz22** [comment]
@bmeluch I changed the release dates to be within two weeks of an nmdc-schema release if there are changes to this repo.

@kheal I edited to make caps and periods consistent. And added the changes for the first release v1.0.0. After this is merged in, I will create a release.

File: CHANGELOG.md
Comment (kheal) [reply CHANGELOG.md:1]: Let's add a section to the CONTRIBUTING.md guide and say that it's for managers only.

Comment (brynnz22) [reply CHANGELOG.md:1]: It says that on line 4

File: .github/RELEASE_GUIDE.md
Comment (brynnz22) [inline .github/RELEASE_GUIDE.md:4]: "This documentation is solely for the maintainers of the nmdc_notebooks repo"

**Pull request merged**
